PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterBubblesTest.java

```diff
@@ -0,0 +1,102 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import nl.tudelft.pl2016gr2.model.Bubble;
+import nl.tudelft.pl2016gr2.model.GraphInterface;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+import nl.tudelft.pl2016gr2.model.PhylogeneticTreeNode;
+import nl.tudelft.pl2016gr2.parser.controller.FullGfaReader;
+import nl.tudelft.pl2016gr2.test.utility.AccessPrivate;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.PriorityQueue;
+
+/**
+ * Tests the FilterBubbles class. The private method buildGraph is not tested,
+ * because by testing the filter method and the getSharedNodes method this 
+ * method is basically already tested. 
+ * @author Casper
+ *
+ */
+public class FilterBubblesTest {
+  
+  private PhylogeneticTreeNode treeRoot = mock(PhylogeneticTreeNode.class);
+  private ArrayList<String> leaves;
+  private OriginalGraph originalGraph;
+  
+  /**
+   * Makes the getLeaves method of the treeRoot return a list with leaves
+   * 'leaf1', 'leaf2', 'leaf3', 'leaf4', and 'leaf5'. Also reads a
+   * test graph from a file. See the file to see what the tree looks like.
+   */
+  @Before
+  public void setup() {
+    leaves = new ArrayList<>();
+    for (int i = 1; i <= 5; i++) {
+      leaves.add("leaf" + i);
+    }
+    
+    when(treeRoot.getLeaves()).thenReturn(leaves);
+    originalGraph = new FullGfaReader("FilterBubblesTest.gfa", 11).getGraph();
```

Comment:
Maybe consider mocking an `OriginalGraph`, since it is a dependency of the class.